crates/omnix-health/src/check/nix_version.rs

```diff
@@ -0,0 +1,54 @@
+use nix_rs::version_spec::NixVersionReq;
+
+use nix_rs::info;
+use serde::{Deserialize, Serialize};
+
+use crate::traits::*;
+
+/// Check that [nix_rs::version::NixVersion] is set to a good value.
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+#[serde(default, rename_all = "kebab-case")]
+pub struct NixVersionCheck {
+    pub supported: String,
```

Comment:
I wish serde supported passing a function as a deserializer. I could just use `pub supported: NixVersionReq` here, if it did. Maybe it does and I haven’t looked properly?

Reply:
Maybe `#[serde(deserialize_with = “<function-name>”)]`?

Reply:
https://github.com/juspay/omnix/pull/398/commits/53355baad21ed7cdaaa0e21f171d8ffb7215f800

works

Reply:
Switched to `DeserializeFromStr`